### PR TITLE
Improve url encoding for Firefox containers

### PR DESCRIPTION
**Pull request by blandir**
Fixes issue #73

### CODE REVIEW

File: src/browser_repository.rs [hunk 1]
```diff
@@ -433,11 +434,10 @@ impl SupportedAppRepository {
...
                 let fake_url = "ext+container:name=".to_string() + container_name;
 
-                // handle https://github.com/honsiorovskyi/open-url-in-container/issues/112
-                let url_encoded = url.replace("&", "%26");
-                let url_encoded = url_encoded.replace("+", "%2B");
+                // encode target url so it can be passed as a parameter
+                let url_encoded = encode(url);
 
-                let full_url = fake_url + "&url=" + url_encoded.as_str();
+                let full_url = fake_url + "&url=" + &*url_encoded;
```
Comment (liias) [inline src/browser_repository.rs:440]: I'd prefer to use `url_encoded.as_ref()` here, just because intent looks a bit more clear.

**Review by liias — CHANGES_REQUESTED** [review]
Whohoo, first pull request!
 
Thank you for your time @blandir! 

Everything works great, I just added two small comments  for personal preferences that I'd like to have for this project.

**Comment by liias** [comment]
Kind of unfortunate timing, but it looks like the crate author is not maintaining that crate anymore. https://github.com/kornelski/rust_urlencoding/issues/6

I'm alright keeping it still. It seems to be used in some larger projects and will probably get a new maintainer. Plus it's small.

**Comment by blandir** [comment]
Unfortunate timing, indeed.

I searched a bit and maybe `use url::form_urlencoded::byte_serialize;` could also work. Here is an [example](https://rust-lang-nursery.github.io/rust-cookbook/encoding/strings.html#encode-a-string-as-applicationx-www-form-urlencoded). Plus we would not need the additional dependency.

I'll do some tests and either update the PR or make the changes you requested.

**Comment by blandir** [comment]
I tested the example from above. An `assert_eq!()` showed that both variants produce the same result.

Please check the udpated PR when you have time.

**Comment by liias** [comment]
Nice, this is even better for maintenance! 
I tested, and it works well. Thank you!

**Pull request merged**